plotting_tools/plot_histograms.C

```diff
@@ -1,93 +1,325 @@
     
     for(auto &[hist_name, params] : hist_names){
-      auto hist = (TH1D*)input_file->Get(hist_name.c_str());
-      auto [logy, logx, rebin, xMin, xMax] = params;
-      
-      if(hist->GetEntries() == 0) continue;
-      
-      hist->SetLineColor(color);
-      hist->Rebin(rebin);
-      hist->Scale(rebin/hist->GetEntries());
-      hist->Sumw2(false);
-      
-      stacks[hist_name]->Add(hist);
-      stacks[hist_name]->SetTitle(hist->GetTitle());
-    
-      if(find(in_legend.begin(), in_legend.end(), file_name) == in_legend.end()){
-        legend->AddEntry(hist, title.c_str(), "l");
-        in_legend.push_back(file_name);
+      for (auto prefix : prefixes){
+        for (auto category : categories) {
+          string full_hist_name = prefix + category + hist_name;
+          auto hist = (TH1D*)input_file->Get(full_hist_name.c_str());
+          auto [logy, logx, rebinned, rebin, xMin, xMax, yMin, yMax, xlabel] = params;
+          
+          cout << file_name << ":  " << full_hist_name << endl;
+          if(hist->GetEntries() == 0) continue;
+          
+          hist->SetLineColor(color);
+          hist->SetLineWidth(2);
+          hist->Rebin(rebin);
+          if(weighted) {hist->Scale(rebin*int_lumi*cross_sec/N_tot);}
+          else{hist->Scale(rebin/hist->GetEntries());}
+          hist->Sumw2(false);
+
+          if(rebinned)
+          {
+            for(int i=0; i<hist->GetNbinsX(); i++){
+              hist->SetBinContent(i+1, hist->GetBinContent(i+1)/(lxy_bins[i+1]-lxy_bins[i]));
+            }
+          }
+          
+          if(signal) {stacks_signal[full_hist_name]->Add(hist);}
+          else {
+            hist->SetFillColor(color);
+            stacks_background[full_hist_name]->Add(hist);
+          }
+        
+          if(find(in_legend.begin(), in_legend.end(), file_name) == in_legend.end()){
+            if(signal){legend->AddEntry(hist, title.c_str(), "l");}
+            else {legend->AddEntry(hist, title.c_str(), "f");}
+            in_legend.push_back(file_name);
+          }
+          input_file->cd();
+        }
       }
     }
   }
-  
-  int i_pad = 1;
+
+  // TODO: make this possible to run with only signal or only background
+  cout << "Saving all plots in: " << output_path << endl;
+  // gROOT->SetBatch(kTRUE);
+  TH1F *ghost_hist;
   for(auto &[hist_name, params] : hist_names){
+    for(auto prefix : prefixes){
+      for (auto category : categories) {
+        string full_hist_name = prefix + category + hist_name;
     
-    auto [logy, logx, rebin, xMin, xMax] = params;
-    
-    canvas->cd(i_pad++);
-    if(logy) gPad->SetLogy();
-    if(logx) gPad->SetLogx();
-    stacks[hist_name]->Draw();
+        auto [logy, logx, rebinned, rebin, xMin, xMax, yMin, yMax, xlabel] = params;
 
-    stacks[hist_name]->GetXaxis()->SetLimits(xMin, xMax);
-    
-    legend->Draw();
+        auto c = new TCanvas("c", "c");
+        c->cd();
```

Comment:
Could we rename it to "canvas", or anything that's searchable and cannot be confused for instance with the speed of light? ;)